src/main/java/com/github/manolo8/darkbot/gui/titlebar/ExtraButton.java

```diff
@@ -31,12 +33,41 @@
     private static final Set<ExtraMenuProvider> EXTRA_DECORATIONS = new LinkedHashSet<>();
     private static final Lazy<Void> clean = new Lazy.NoCache<>();
 
-    public static void setExtraDecorations(Stream<ExtraMenuProvider> provider) {
+    private static final PluginExtraMenuProvider PLUGIN_DECORATIONS = new PluginExtraMenuProvider();
+    private static final Map<Plugin, Set<ExtraMenuProvider>> plugins = new HashMap<>();
```

Comment:
This shouldn't be needed here at all, you have a list of extra menu providers, make those providers have a default one and a plugin one, both of them installed natively, and on setExtraDecorations filter out the plugin-specific ones and add them to the native plugin extra menu provider